### ISSUE TITLE
Combining Suppress with another annotation doesn't suppress warnings 

### ISSUE THREAD

**Issue by vovahost** [detekt/detekt#5249]
Consider the following code:
```
@CombinedAnnotation("very long text which exceeds MaxLineLength")
@Suppress("MaxLineLength")
fun myFunction() {
    // Empty
}
```
The above `Suppress("MaxLineLength")` works as expected.

Now if I move the `@Suppress("MaxLineLength")` to `@CombinedAnnotation`, the suppression doesn't work anymore:
```

@Retention(AnnotationRetention.SOURCE)
@Target(AnnotationTarget.FUNCTION)
@Repeatable
@Suppress("MaxLineLength")
annotation class CombinedAnnotation(
    vararg val value: String,
)

@CombinedAnnotation("very long text which exceeds MaxLineLength")
fun myFunction() {
    // Empty
}
```

## Expected Behavior
Detekt should suppress the `MaxLineLength` rule.

## Observed Behavior
Detekt DOESN'T suppress the `MaxLineLength`  rule.

## Steps to Reproduce
Just copy the above code.

## Context
I'm defining the `CombinedAnnotation` which will be used in the entire codebase. I want to exclude the `MaxLineLength` check when I'm using this annotation.
Looks like there's no `ignoreAnnotation` for `MaxLineLength` rule as well.

## Your Environment
* Version of detekt used: 1.21.0
* Version of Gradle used (if applicable): 7.5
* Kotlin: 1.7.10

**Comment by cortinico** [comment]
That's not how Annotations in Java generally works.

The fact that you annotate `CombinedAnnotation` with `@Suppress` doens't imply that the `@Suppress` annotation should be propagated to usages of `CombinedAnnotation`.

Implementing this will also have side effects, as I can imagine scenarios where you might actually want to suppress an inpsection **only** on declaration of an annotation and not on every usage.

**Comment by BraisGabin** [comment]
As @cortinico said this is not how the suppression is meant to work and it could create other problems to allow that.

I think that you could fix your use case adding this in your configuration file:

```yml
MaxLineLength:
  ignoreAnnotated:
    - 'CombinedAnnotation'
```

**Comment by vovahost** [comment]
@BraisGabin That would be ideal if `MaxLineLength` rule supported `ignoreAnnotated` option https://detekt.dev/docs/rules/style#maxlinelength
Is it possible to add support for `ignoreAnnotated`?

**Comment by BraisGabin** [comment]
If it doesn't, its a bug. Could you try?

**Comment by vovahost** [comment]
@BraisGabin I tried and it didn't work. That's the reason why I opened this issue.
Also it doesn't say in the documentation that it's supported https://detekt.dev/docs/rules/style#maxlinelength.
You can double check as well.

**Comment by cortinico** [comment]
> Also it doesn't say in the documentation that it's supported https://detekt.dev/docs/rules/style#maxlinelength.

Please note that `ignoreAnnotated` is a feature that applies to all the rules. It's documented here: https://detekt.dev/docs/introduction/suppressors

**Comment by vovahost** [comment]
@cortinico I've tested it in the latest Android Studio Electric Eel and it works as you described. It still doesn't work for some reason in Intellij Idea.
If I'm running the task `detekt` it throws the error:
```
Property 'formatting' is misspelled or does not exist.
```
I'll test a bit more.

**Comment by BraisGabin** [comment]
If I'm running the task `detekt` it throws the error:
> 
> ```
> Property 'formatting' is misspelled or does not exist.
> ```

That seems like in your `.yml` configuration file you have the key `formatting:` but you don't apply the plugin `detekt-formatting`. Old vrsions of detekt auto generate that key automatically even when you don't us `detekt-formatting` so it should be safe to remove that key from your configuration file.
